allennlp/modules/token_embedders/elmo_token_embedder.py

```diff
@@ -0,0 +1,219 @@
+        ret = self._projection(ret)
+
+        # reshape to (batch_size, sequence_length, embedding_dim)
+        batch_size, sequence_length, _ = inputs.data.shape
+
+        return {
+            'mask': mask_with_bos_eos,
+            'token_embedding': ret.view(batch_size, sequence_length + 2, -1)
+        }
+
+    def _load_weights(self):
+        self._load_char_embedding()
+        self._load_cnn_weights()
+        self._load_highway()
+        self._load_projection()
+
+    def _load_char_embedding(self):
+        with h5py.File(cached_path(self._weight_file), 'r') as fin:
+            char_embed_weights = fin['char_embed'][...]
+
+        weights = numpy.zeros(
+            (char_embed_weights.shape[0] + 1, char_embed_weights.shape[1]),
+            dtype=DTYPE
+        )
+        weights[1:, :] = char_embed_weights
+
+        self._char_embedding_weights = torch.nn.Parameter(
+            torch.FloatTensor(weights), requires_grad=False
+        )
+
+    def _load_cnn_weights(self):
+        cnn_options = self._options['char_cnn']
+        filters = cnn_options['filters']
+        char_embed_dim = cnn_options['embedding']['dim']
+
+        convolutions = []
+        for i, (width, num) in enumerate(filters):
+            conv = torch.nn.Conv1d(
+                in_channels=char_embed_dim,
+                out_channels=num,
+                kernel_size=width,
+                bias=True
+            )
+            # load the weights
+            with h5py.File(cached_path(self._weight_file), 'r') as fin:
+                w = fin['CNN']['W_cnn_{}'.format(i)][...]
+                b = fin['CNN']['b_cnn_{}'.format(i)][...]
+
+            w_reshaped = numpy.transpose(w.squeeze(axis=0), axes=(2, 1, 0))
+            if w_reshaped.shape != tuple(conv.weight.data.shape):
+                raise ValueError("Invalid weight file")
+            conv.weight.data.copy_(torch.FloatTensor(w_reshaped))
+            conv.bias.data.copy_(torch.FloatTensor(b))
+
+            conv.weight.requires_grad = False
+            conv.bias.requires_grad = False
+
+            convolutions.append(conv)
+
+        self._convolutions = convolutions
+
+    def _load_highway(self):
+        # the highway layers have same dimensionality as the number of cnn filters
+        cnn_options = self._options['char_cnn']
+        filters = cnn_options['filters']
+        n_filters = sum(f[1] for f in filters)
+        n_highway = cnn_options['n_highway']
+
+        # create the layers, and load the weights
+        self._highways = Highway(n_filters, n_highway, activation=torch.nn.functional.relu)
+        for k in range(n_highway):
+            # The AllenNLP highway is one matrix multplication with concatenation of
+            # transform and carry weights.
+            with h5py.File(cached_path(self._weight_file), 'r') as fin:
+                W_transform = fin['CNN_high_{}'.format(k)]['W_transform'][...].T
```

Comment:
I'd strongly prefer using `numpy.transpose()` rather than `.T`, it's so easy to miss. Also if we could use lowercase full name variables that would be great, `W_transform` -> `weight_transform`.

Reply:
Also if you could add a top level comment about why we have to transpose the weights, that would be great.

Reply:
I wanted to be consistent with the other place I have similar code for naming.  It will make maintaining this much easier.